Add findAndModify command

scottmessinger opened the pull request.

ericmj commented on lib/mongo.ex:
```diff
@@ -116,6 +116,56 @@ defmodule Mongo do
 
   @doc """
+  The `find_and_modify` command modifies and returns a single document.
+  By default, the returned document does not include the modifications made on the update.
```
Insert newline here so we a separate paragraph in the docs.

ericmj commented on lib/mongo.ex:
```diff
@@ -116,6 +116,56 @@ defmodule Mongo do
+
+    opts = Keyword.drop(opts, ~w(new fields remove upsert bypass_document_validation write_concern max_time)a)
+
+    with {:ok, doc} <- command(conn, query, opts), do: {:ok, doc}
```
The `with` does not have any effect.

ericmj commented on lib/mongo.ex:
```diff
@@ -116,6 +116,56 @@ defmodule Mongo do
+    bypassDocumentValidation: boolean,
+    writeConcern: %{}
+  ]
+  @spec find_and_modify(conn, collection, BSON.document, find_and_modify_opts) :: %{value: %{}, lastErrorObject: %{}, ok: non_neg_integer}
```
The spec does not seem to match the actual return value. The returned map should be converted to atom keys and the value is wrapped in `{:ok, ...}`.

Please add a bang-version of this function (`find_and_modify!`). Check the other functions for examples of the typespecs.

scottmessinger committed:
Add findAndModify commands

Commands are:
* find_one_and_update
* find_one_and_replace
* find_one_and_delete

See:
https://github.com/mongodb/specifications/blob/master/source/crud/crud.r
st#find-and-modify

Note:
Attempt to make find_and_modify fail on 2.4 (to make the bangified
tests pass) but 2.4 won't fail where other versions will.

scottmessinger committed:
Update mix.lock

Updated connection package
